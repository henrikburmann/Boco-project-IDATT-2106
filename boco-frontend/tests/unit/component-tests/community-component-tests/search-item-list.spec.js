import { shallowMount } from "@vue/test-utils";
import SearchItemListComponent from "@/components/ItemComponents/SearchItemList.vue";

describe("SearchItemListComponent elements rendering", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(SearchItemListComponent);
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it("Tests setting values of input field", async () => {
    const searchInput = wrapper.find("#searchInput");
    await searchInput.setValue("Dyson");
    expect(searchInput.element.value).toBe("Dyson");
  });
});
