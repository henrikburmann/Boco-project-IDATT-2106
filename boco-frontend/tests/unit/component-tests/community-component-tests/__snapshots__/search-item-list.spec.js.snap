// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItemListComponent elements rendering renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- A template for searching in item list -->
  <section
    class="relative w-full max-w-md px-5 py-4 mx-auto rounded-md"
  >
    <div
      class="relative"
      id="searchComponent"
    >
      <span
        class="absolute inset-y-0 left-0 flex items-center pl-3"
      >
        <svg
          class="w-5 h-5 text-gray-400"
          fill="none"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 21L15 15M17 10C17 13.866 13.866 17 10 17C6.13401 17 3 13.866 3 10C3 6.13401 6.13401 3 10 3C13.866 3 17 6.13401 17 10Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </span>
      <input
        class="w-full py-3 pl-10 pr-4 text-gray-700 bg-white border rounded-md dark:bg-gray-800 dark:text-gray-300 dark:border-gray-600 focus:border-primary-medium dark:focus:border-primary-medium focus:outline-none focus:ring"
        id="searchInput"
        placeholder="SÃ¸k"
        type="text"
      />
    </div>
    <div
      class="absolute inset-x-0 px-6 py-3 mt-4 border-2 border-slate-500"
    >
      <div
        class="grid grid-cols-2"
      >
        
        <item-card-stub
          item="[object Object]"
        />
        <item-card-stub
          item="[object Object]"
        />
        <item-card-stub
          item="[object Object]"
        />
        
      </div>
    </div>
  </section>
  
</div>
`;
